### ISSUE TITLE
About correlation layer.

### ISSUE THREAD

**Issue by MeLonJ10** [CVLAB-Unibo/Real-time-self-adaptive-deep-stereo#72]
Hi @AlessioTonioni, thanks for sharing your work! 
I have a question about correlation layer in tf.
`def correlation_tf(x, y, max_disp, stride=1, name='corr'):
	with tf.variable_scope(name):
		corr_tensors = []
		y_shape = tf.shape(y)
		y_feature = tf.pad(y,[[0,0],[0,0],[max_disp,max_disp],[0,0]])
		for i in range(-max_disp, max_disp+1,stride):
			shifted = tf.slice(y_feature, [0, 0, i + max_disp, 0], [-1, y_shape[1], y_shape[2], -1])
			corr_tensors.append(tf.reduce_mean(shifted*x, axis=-1, keepdims=True))

		result = tf.concat(corr_tensors,axis=-1)
		return result`
Does the correlation layer defined in this way mean that, it only use single pixel to calculate correlation rather than a patch centered at the candidate pixels?
Thank you so much!

**Comment by AlessioTonioni** [comment]
Hi, 
Yes the correlation layer computes correlation between single locations of the volumes provided as inputs.
In our architecture  it is applied at lower scales in the netwrok and on the output of convolutional layers, therefore each location contains information coming from all the receptive field of the previous convolutional layers. For this reason the layer operates on single locations but it can be seemn as computing correlation between patches.
